File: src/fizzbuzz.pl
foreach my $i (1..100) {
	if(($i % 3 == 0) && ($i % 5 == 0)) {
		print "fizzbuzz\n";
		} elsif($i % 3 == 0) {
			print "fizz\n";
			} elsif($i % 5 == 0) {
				print "buzz\n";
				} else {
					print $i;
					print "\n";
				}
			}

File: src/fizzbuzz.hs
main = mapM_ putStrLn $ map fizzbuzz [1..100]

fizzbuzz :: (Num a, Show a) => a -> String
fizzbuzz n
	| n `mod` 15 == 0 = "fizzbuzz"
	| n `mod` 3  == 0 = "fizz"
	| n `mod` 3  == 0 = "buzz"
	| otherwise = show n
